react/components/Collapsible.js

```diff
@@ -1,51 +1,76 @@
 import React from 'react'
-import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import withScrollAnimation from './ScrollAnimation'
-import { Border } from './Helpers'
+import { NoSSR } from 'vtex.render-runtime'
+import { useSpring, animated, config as springPresets } from 'react-spring'
 import useDevice from '../hooks/useDevice'
+import useScrollDirection from '../hooks/useScrollDirection'
+import Border from './Helpers/Border'
+import { collapsible, lean } from '../defaults'
 
-import header from '../store-header.css'
+import styles from '../store-header.css'
 
-const Collapsible = ({
-  children,
-  leanMode,
-  animation,
-  didAnimate,
-  onAnimate,
-}) => {
+/**
+ * Represents a collapsible part of the header
+ */
+const Collapsible = ({ children, leanMode, collapsibleAnimation }) => {
   const { desktop } = useDevice()
-  onAnimate(didAnimate)
+  const { scroll, scrollingUp } = useScrollDirection()
+  const {
+    onScroll,
+    anchor,
+    always,
+    from,
+    to,
+    preset,
+    config,
+  } = collapsibleAnimation
+
+  let animationTrigger = true
+
+  if (onScroll) {
+    const animateWhen = scroll < anchor
+    const elastic = always ? scrollingUp : false
+    animationTrigger = animateWhen || elastic
+  }
+
+  const props =
```

Comment:
Change it to `animatedStyle`. At a first glance, I confused with actually props from this component.

Reply:
I agree, good suggestion.